Variables and tabs bottom

gkalvatcheva opened the pull request:
- color variables
- mixins
- bottoms tabs style

gkalvatcheva committed:
- set up color variables

gkalvatcheva committed:
- font awesome install
- variable replacement

gkalvatcheva committed:
- import font google font
- set font for app to Open Sans

gkalvatcheva committed:
- added wrapper around views tabs
- added wrapper around charts tab
- view mode tabs styling
- water point status name
- height variable adjustment
- general link style
- transition and tab text mixins

uniphil commented on app/scripts/components/boilerplate/view-mode.jsx:
```diff
@@ -7,6 +7,7 @@ require('stylesheets/boilerplate/view-mode');
 const ViewMode = React.createClass({
   render() {
     return (
+      <div className="view-mode-wrapper">
```
lets make this just `view-mode` and rename or remove the `className` from the `<ul>`. Can we reindent the `<ul` and it's contents?

uniphil commented on app/scripts/components/dashboard/charts-container.jsx:
```diff
@@ -12,7 +12,7 @@ const ChartsContainer = React.createClass({
       <div className="charts-container">
         <div className="charts-container-header">
-          <T k="charts.toggle.activate" />
+          <span className="tab-label"><T k="charts.toggle.activate" /></span>
```
@esoliani do you think we should let `<T>` take a `className` prop and put it on the `<span>` in `T` that holds the text? Or should we keep it simple and wrap it in a span like this when we want class names?

uniphil commented on app/stylesheets/boilerplate/footer.scss:
```diff
@@ -3,10 +3,10 @@
 .footer {
   background: $theme-blue;
-  color: $white;
+  color: $bg-color;
```
:+1:

uniphil commented on app/stylesheets/boilerplate/view-mode.scss:
```diff
@@ -1,12 +1,38 @@
+  width: 100%;
+}
+
+ul.view-mode {
```
So, if the `.view-mode-wrapper` class just becomes `.view-mode` (see comment above), lets nest this `ul` inside it, and use the selector `& > ul`.

uniphil commented on app/stylesheets/boilerplate/view-mode.scss:
```diff
@@ -1,12 +1,38 @@
+  table-layout: fixed;
+  width: 100%;
 
   li {
```
Lets do `& > li` here. (I know, I wrote this li selector I'm now complaining about :smile: )

galinagk replied:
No worries. Added.

uniphil commented on app/stylesheets/config/type.scss:
```diff
@@ -1,3 +1,27 @@
+
+// basic text style in top and bottom tabs
+@mixin tab-text() {
+  font-size: 12px;
```
Lets put `font-size` as a variable. By "top and bottom tabs" do you mean the filters? In the screenshot they look like `font-weight: normal` I think, so maybe it's better to just use variables instead of a mixin

uniphil commented on app/stylesheets/config/type.scss:
```diff
@@ -1,3 +1,27 @@
+}
+
+@mixin transitionAll() {
+  -moz-transition: all 250ms ease;
```
Lets leave out browser prefixes from our styles. We can insert them automatically with a processing step on build.

So then, we can just put the transition rule right into the styles instead of using a mixin.

Maybe the duration should be a variable in a `config/timing.scss` file?

uniphil commented on app/stylesheets/config/type.scss:
```diff
@@ -1,3 +1,27 @@
+  transition: all 250ms ease;
+}
+
+body,
```
Everything in `config/` should have only mixins and variables. Maybe we need a `stylesheets/global.scss` file for rules like this.

galinagk replied:
Good idea. Added global for the links, global font settings, etc

uniphil commented on app/stylesheets/dashboard/charts-container.scss:
```diff
@@ -1,5 +1,16 @@
   background: $grey-light;
 }
+
+.tab-label {
```
this class is not used anywhere

galinagk replied:
Right now it's used in one place in charts-container.jsx for the charts toggle label

uniphil replied:
my mistake, sorry!

Ok great. Move `.tab-label` inside `charts-container {` then and we're good.

uniphil commented on package.json:
```diff
@@ -71,6 +71,7 @@
     "leaflet": "0.7.5",
     "object-assign": "4.0.1",
     "react": "0.13.3",
+    "react-font-awesome": "https://github.com/janmyler/react-font-awesome/tarball/261c1719581e9125a544f8b5b9efc2e27073d970",
```
:+1:

uniphil commented on app/stylesheets/dashboard/charts-container.scss:
```diff
@@ -1,5 +1,16 @@
+  background: $grey-light;
+  color: $text-color;
+  display: table-cell;
+  min-width: 200px;
```
Maybe move this `200px` to a variable in `dimensions.scss`?

gkalvatcheva committed:
- nested tab label in charts container

gkalvatcheva committed:
- new global stylesheet
- new variable for chart tab width
- new var for font size in labels (replaced the mixin)

gkalvatcheva committed:
- charts image for charts tab
- new variable for icon
- variable for link transitions

gkalvatcheva committed:
- char position in charts tab

uniphil commented on app/stylesheets/dashboard/charts-container.scss:
```diff
@@ -1,5 +1,25 @@
 .charts-container {
   background: $grey-light;
+
+  .tab-label {
```
lets make this selector more specific, since `.charts-container` will have lots of children and we don't want these rules to leak into them.
Maybe `& > .charts-container-header > .tab-label {`

uniphil commented on app/scripts/components/dashboard/charts-container.jsx:
```diff
@@ -12,7 +12,8 @@ const ChartsContainer = React.createClass({
         <div className="charts-container-header">
-          <T k="charts.toggle.activate" />
+          <span className="tab-label">
+          <T k="charts.toggle.activate" /></span>
```
Watch the indentation here -- `<T>` should be indented inside `<span>`, and the closing `</span>` should be on a new line.

uniphil commented on app/stylesheets/global.scss:
```diff
@@ -0,0 +1,12 @@
+}
+
+a {
+  @include link-transition( 600ms, ease);
```
Lets either
a) make `link-transition` have default values for timing (imported from `timing`) so that here we would just have `link-transiton()`, **or**
b) drop the `link-transition` mixin and just use variables with regular `transition`, like
`transition: color $link-duration $link-easing;`.

gkalvatcheva replied:
I like the 2nd option since we won't need to have an extra scss file for timing. :) Or should we still keep it? I'm not sure we'll have more variables there.

uniphil replied:
yeah good point, lets drop it for now. We can pull out the timings again later if we want.

uniphil commented on app/stylesheets/config/type.scss:
```diff
@@ -1,3 +1,11 @@
 }
+
+@mixin link-transition($link-duration, $link-easing) {
+  transition: $link-duration, $link-easing;
```
I think this needs `color` or `all` before `$link-duration`, if we keep this mixin (see comment in global.scss).

gkalvatcheva committed:
Merge branch 'master' into variables-and-tabs-bottom

gkalvatcheva committed:
Revert "Merge branch 'master' into variables-and-tabs-bottom"

This reverts commit cc2e48677e506001ba4140f2ba68f0f835a2f97a, reversing
changes made to da61656d93142f14d07af5b09a3c953827344f2c.

gkalvatcheva committed:
- removed timing stylesheet
- changes transition mixing to a simple global mixin. I think we’ll use
this for all links and we probably don’t need any other transitions
- added specificity for charts tab

gkalvatcheva committed:
- removed timing import in global stylesheet

gkalvatcheva committed:
-removed view mode wrapper and adjusted styles for this section

gkalvatcheva committed:
- some css linting

uniphil commented:
Manually merged via a91255b6fb219b3efd43405d7233ab3527fb768f
